tests/testthat/test-shinytest2-tm_a_pca.R

```diff
@@ -0,0 +1,245 @@
+app_driver_tm_a_pca <- function() {
+  # Dataset only used once
+  data <- within(teal.data::teal_data(), {
+    require(nestcolor)
+
+    USArrests <- USArrests # nolint: object_name.
+  })
+  teal.data::datanames(data) <- "USArrests"
+
+
+  init_teal_app_driver(
+    data = data,
+    modules = tm_a_pca(
+      dat = teal.transform::data_extract_spec(
+        dataname = "USArrests",
+        select = teal.transform::select_spec(
+          choices = teal.transform::variable_choices(
+            data = data[["USArrests"]],
+            c("Murder", "Assault", "UrbanPop", "Rape")
+          ),
+          selected = c("Murder", "Assault"),
+          multiple = TRUE
+        )
+      ),
+      size = c(3, 1, 5),
+      alpha = c(.5, 0, 1),
+      font_size = c(10, 8, 15),
+      ggtheme = "light",
+      rotate_xaxis_labels = TRUE,
+      pre_output = shiny::tags$div(id = "unique_id_pre", "A pre output"),
+      post_output = shiny::tags$div(id = "unique_id_post", "A post output")
+    )
+  )
+}
+
+# Defaults --------------------------------------------------------------------
+
+testthat::test_that("e2e - tm_a_pca: module is initialised with the specified defaults in function call", {
```

Comment:
Adds a simple test to make sure that all the non-default options are implemented in the encoding.